database/migrations/000108_data_sources.up.sql

```diff
@@ -0,0 +1,58 @@
+-- SPDX-FileCopyrightText: Copyright 2024 The Minder Authors
+-- SPDX-License-Identifier: Apache-2.0
+
+BEGIN;
+
+-- This migration adds storage support for data sources. The only
+-- constraints we enforce at the database layer are
+--
+-- * functions can only reference one data source, and must be deleted
+--   if the data source is deleted
```

Comment:
I think I missed this in the design -- how do functions and data sources interact?

Reply:
The idea is that a data source is a "container" of functions that also defines a namespace.
As an example, this might be a shortened version of Trusty API v2 data source
```yaml
version: v1
type: data-source
name: trusty
context:
  project: foo
rest:
 summary:
    endpoint: "https://api.trustypkg.dev/v2/summary?package_name={package_name}
    method: GET
    args:
      - package_name:
        schema:
          type: string
 pkg:
    endpoint: "https://api.trustypkg.dev/v2/pkg?package_name={package_name}"
    method: GET
    args:
      package_name:
        schema:
          type: string
```

This YAML defines data source `trusty` having "functions" `summary` and `pkg`. Eventually, they would be available to Rego engine as `minder.datasource.trusty.summary(...)` and `minder.datasource.trusty.pkg(...)`. Some field names might be different, and some lists could be maps and vice versa, but that's about it.

Getting back to the comment, we thought that saving all that informatino in a single row of table `data_sources` was too coarse, and decided to add a separate table containing functions. I'm not sure I answered your question.

Reply:
Thanks, the last time I'd reviewed the doc, multiple endpoints for the same data source weren't in it. 

The tables make sense to me.